src/taskcreator.js

```diff
@@ -79,7 +93,34 @@ class TaskCreator {
       this.lastCreateTask = {taskId: options.taskId, task};
       return {status: {taskId: options.taskId}};
     }
-    return await queue.createTask(options.taskId, task);
+
+    let lastFire, taskCreateRes;
+    try {
+      taskCreateRes = await queue.createTask(options.taskId, task);
+      lastFire = {
+        result: 'success',
+        taskId: hook.nextTaskId,
```

Comment:
This is different from the taskId that was just created!  I think you want `options.taskId` here.  You'll also need the taskId even in the failure case (since taskId will be part of the rowKey).